src/websockets/server.py

```diff
@@ -714,7 +714,12 @@ async def _close(self) -> None:
         # asyncio.wait doesn't accept an empty first argument
         if self.websockets:
             await asyncio.wait(
-                [websocket.close(1001) for websocket in self.websockets],
+                [
+                    asyncio.ensure_future(websocket.close(1001))
+                    if sys.version_info[:2] < (3, 7)
+                    else asyncio.create_task(websocket.close(1001))
+                    for websocket in self.websockets
+                ],
```

Comment:
Let's not add a one-off implementation of this compatibility requirement just in one place.

When we want to support Python 3.9, let's add a compatibility  function, similar to what db25f49496343bb6aacbe31994da83a5470dc67c removed.

Reply:
Strictly speaking, for the purposes of this issue, it's not really necessary to use `asyncio.create_task()` (since `ensure_future()` hasn't actually been deprecated). So this issue could also be addressed simply by using `ensure_future()` and not adding a compatibility layer.

In terms of a compatibility layer, I'm not sure if having the ability to use `create_task()` in higher versions outweighs the maintenance burden of the compat function, as well introducing different code paths (and potentially different behavior) for different versions.

Reply:
Indeed — I didn't realize that ensure_future still worked on Python 3.9.

The Python docs only say that create_task is "higher-level" than ensure_future. I don't expect there will be a noticeable difference for our usage.

Like Chris, I'd be inclined to stick with ensure_future until all versions of Python supported by websockets support the create_task API, then switch from ensure_future to create_task.

Reply:
👍 The other relevant statement in the docs is in the [`ensure_future()`](https://docs.python.org/3/library/asyncio-future.html#asyncio.ensure_future) docs:

> **Important:** See also the create_task() function which is the preferred way for creating new Tasks.

But nevertheless, like you suggest, I'm not sure it confers any particular advantage..